packages/unplugin-typegpu/test/auto-naming.test.ts

```diff
@@ -0,0 +1,427 @@
+
+      console.log(bindGroupLayout, vertexLayout);
+    `;
+
+    expect(babelTransform(code)).toMatchInlineSnapshot(`
+      "import tgpu from 'typegpu';
+      import * as d from 'typegpu/data';
+      const bindGroupLayout = (globalThis.__TYPEGPU_AUTONAME__ ?? (a => a))(tgpu.bindGroupLayout({}), "bindGroupLayout");
+      const vertexLayout = (globalThis.__TYPEGPU_AUTONAME__ ?? (a => a))(tgpu.vertexLayout(n => d.arrayOf(d.u32, n)), "vertexLayout");
+      console.log(bindGroupLayout, vertexLayout);"
+    `);
+  });
+
+  it(`works with tgpu['~unstable'] items`, async () => {
+    const code = `\
+      import tgpu from 'typegpu';
+      import * as d from 'typegpu/data';
+
+      let nothing, accessor = tgpu['~unstable'].accessor(d.u32);
+      let shell = tgpu['~unstable'].fn([]);
+      var fn = tgpu['~unstable'].fn([])(() => {});
+      const cst = tgpu['~unstable'].const(d.u32, 1);
+
+      console.log(accessor, shell, fn, cst);
+    `;
+
+    expect(babelTransform(code)).toMatchInlineSnapshot(`
+      "import tgpu from 'typegpu';
+      import * as d from 'typegpu/data';
+      let nothing,
+        accessor = (globalThis.__TYPEGPU_AUTONAME__ ?? (a => a))(tgpu['~unstable'].accessor(d.u32), "accessor");
+      let shell = tgpu['~unstable'].fn([]);
+      var fn = (globalThis.__TYPEGPU_AUTONAME__ ?? (a => a))(tgpu['~unstable'].fn([])(($ => (globalThis.__TYPEGPU_META__ ??= new WeakMap()).set($.f = () => {
+        throw new Error("The function \\"<unnamed>\\" is invokable only on the GPU. If you want to use it on the CPU, mark it with the \\"kernel & js\\" directive.");
+      }, {
+          v: 1,
+          ast: {"params":[],"body":[0,[]],"externalNames":[]},
+          externals: {},
+        }) && $.f)({})), "fn");
+      const cst = (globalThis.__TYPEGPU_AUTONAME__ ?? (a => a))(tgpu['~unstable'].const(d.u32, 1), "cst");
+      console.log(accessor, shell, fn, cst);"
+    `);
+  });
+
+  it('works with structs', async () => {
+    const code = `\
+      import * as d from 'typegpu/data';
+      import { struct } from 'typegpu/data';
+
+      const myStruct1 = d.struct({ a: d.u32 });
+      const myStruct2 = struct({ a: u32 });
+      const bait = d.i32(1);
+
+      console.log(myStruct1, myStruct2, bait);
+    `;
+
+    expect(babelTransform(code)).toMatchInlineSnapshot(`
+      "import * as d from 'typegpu/data';
+      import { struct } from 'typegpu/data';
+      const myStruct1 = (globalThis.__TYPEGPU_AUTONAME__ ?? (a => a))(d.struct({
+        a: d.u32
+      }), "myStruct1");
+      const myStruct2 = (globalThis.__TYPEGPU_AUTONAME__ ?? (a => a))(struct({
+        a: u32
+      }), "myStruct2");
+      const bait = d.i32(1);
+      console.log(myStruct1, myStruct2, bait);"
+    `);
+  });
+
+  it('works with root items', async () => {
+    const code = `\
+      import tgpu from 'typegpu';
+
+      const root = await tgpu.init();
+      const myBuffer = root.createBuffer(d.u32, 2);
```

Comment:
i'd add test for when there are multiple chained methods. so like when creating buffer usages with .$usage, or setting externals on a function